### DISCUSSION TITLE
Nicely documented OSS projects made on top on uWS.js?

### DISCUSSION THREAD

**Discussion by harshmandan** [uNetworking/uWebSockets.js#425] [General]
Hello,
I am currently using express.js & socket.io on my production app and now when I have real traffic on my app, I'm going to the max optimization route.

But a lot of my code is dependent on express modules - I use EJS, middlewares, multer, express-sessions, etc.

I know there are some projects that simplify uws.js like: siffr-server & nanoexpress. But when I'm going the route to rewriting my code I do not want another dependency layer.

Are there any good projects/repository which can offer a good starting point? A good project with all these things already handled would help. Things like Router, file-upload, multipart requests, sessions handling, middlewares and more.

Thanks!

**Comment by ghost** [comment]
Hello, I don't know any such project. There are a few larger-ish projects here and there on GitHub, but uWS is way more niche than something like Express or Socket.IO. Most users are companies that develop proprietary services directly using uWS, so there is very little openly available information outside of this repo.

**Reply by harshmandan** [reply]
If that is the case then it becomes difficult to get into. A good boilerplate example for all of the above mentioned things, in a "preferred/opinionated/performant" way by the original developers would not just go a long way, it would help beginners and people looking to switch over alike.

**Comment by harshmandan** [comment]
Found one such good open source project:
https://github.com/ParabolInc/parabol It uses few more parts along with uWS.js

Also a nicle article by the author of the repo to go along with it:
https://dev.to/mattkrick/replacing-express-with-uwebsockets-48ph

**Comment by dalisoft** [comment]
@harshmandan Try [this](https://github.com/nanoexpress/nanoexpress)

**Comment by roelvan** [comment]
I think Deepstream is using UWS under the hood too: https://deepstream.io/ (https://github.com/deepstreamIO/deepstream.io)

**Reply by ghost** [reply]
Yes, I have contracted for them but I'm pretty sure they are out of business nowadays.
